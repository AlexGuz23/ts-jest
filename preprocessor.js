const tsc = require('typescript');
const {getTSConfig} = require('./utils');

module.exports = {
  process(src, path, config) {
    if (path.endsWith('.ts') || path.endsWith('.tsx')) {
      const transpiled = tsc.transpileModule(
        src,
        {
          compilerOptions: getTSConfig(config.globals),
          fileName: path
        });

      const modified = `require('ts-jest').install({environment: 'node', emptyCacheBetweenOperations: true});${transpiled.outputText}`;

      return modified;
    }

    return src;
  }
};