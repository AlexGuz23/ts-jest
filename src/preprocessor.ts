import * as tsc from 'typescript';
import { getTSConfig } from './utils';

export function process(src, path, config) {
  if (path.endsWith('.ts') || path.endsWith('.tsx')) {
    const transpiled = tsc.transpileModule(
      src,
      {
        compilerOptions: getTSConfig(config.globals),
        fileName: path
      });

    const modified = `require('ts-jest').install();${transpiled.outputText}`;

    return modified;
  }

  return src;
}