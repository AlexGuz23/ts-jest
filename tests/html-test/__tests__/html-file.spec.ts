import {process} from '../../../preprocessor';
const config = require('../package.json');

test('transforms html if config.globals.__TRANSFORM_HTML__ is set', () => {
  const source =
`<div class="html-test">
  <span class="html-test__element">This is element</span>
</div>`;
  const path = '/path/to/file.html';

  expect(process(source, path, config.jest)).toMatchSnapshot();
  delete config.jest.globals.__TRANSFORM_HTML__;
  expect(process(source, path, config.jest)).toMatchSnapshot();
});
