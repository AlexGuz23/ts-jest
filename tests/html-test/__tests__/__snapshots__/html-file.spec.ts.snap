// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transforms html if config.globals.__TRANSFORM_HTML__ is set 1`] = `
"module.exports=\`<div class=\\"html-test\\">
  <span class=\\"html-test__element\\">This is element</span>
</div>\`;"
`;

exports[`transforms html if config.globals.__TRANSFORM_HTML__ is set 2`] = `
"<div class=\\"html-test\\">
  <span class=\\"html-test__element\\">This is element</span>
</div>"
`;
